standard/interfaces.md

```diff
@@ -305,75 +405,182 @@ These rules ensure that any covariant or contravariant usage of the interface re
 
-The type of an interface event shall be input-safe.
+- *event_modifier* shall not include `override`.
+- A derived interface may implement an abstract interface event declared in a base interface (§15.8.5).
+- It is a compile-time error for *variable_declarators* in an instance *event_declaration* to contain any *variable_initializer*s.
+- An instance event with the `virtual` or `sealed` modifiers must declare accessors. It is *never* considered to be an automatically implemented field-like event ([§15.8.2](classes.md#1582-field-like-events)).
+- An instance event with the `abstract` modifier must not declare accessors.
+- The type of an interface event shall be input-safe.
 
 ### 19.4.5 Interface indexers
 
-Interface indexers are declared using *interface_indexer_declaration*s:
+This clause augments the description of indexers in classes [§15.9](classes.md#159-indexers) for indexers declared in interfaces.
 
-```ANTLR
-interface_indexer_declaration
-    : attributes? 'new'? type 'this' '[' parameter_list ']'
-      '{' interface_accessors '}'
-    | attributes? 'new'? ref_kind type 'this' '[' parameter_list ']'
-      '{' ref_interface_accessor '}'
-    ;
-```
+Interface indexers are declared using *indexer_declaration*s ([§15.9](classes.md#159-indexers)), with the following additional rules:
+
+- *indexer_modifier* shall not include `override`.
+- An *indexer_declaration* that has an *expression body* or contains an accessor with a body is an implementation ([§19.1](interfaces.md#191-general)), so it is *not* abstract.
+- All the parameter types of an interface indexer shall be input-safe ([§19.2.3.2](interfaces.md#19232-variance-safety)).
+- Any output or reference parameter types shall also be output-safe.
+
+  > *Note*: Output parameters are required to be input-safe due to common implementation restrictions. *end note*
+
+- The type of an interface indexer shall be output-safe if there is a get accessor, and shall be input-safe if there is a set accessor.
+
+### §interface-operators Interface operators
+
+This clause augments the description of *operator_declaration* members in classes [§15.10](classes.md#1510-operators) for operators declared in interfaces.
+
+An *operator_declaration* in an interface is the implementation ([§19.1](interfaces.md#191-general)).
+
+It is a compile-time error for an interface to declare a conversion, equality, or inequality operator.
+
+### §interface-static-constructors Interface static constructors
+
+This clause augments the description of static constructors in classes [§15.12](classes.md#1512-static-constructors) for static constructors declared in interfaces.
+
+The static constructor for a closed interface executes at most once in a given application domain. The execution of a static constructor is triggered by the first of the following actions to occur within an application domain:
```

Comment:
Where is the term "closed interface" defined? I'm slightly concerned that this language will conflict with the "closed hierarchies" proposal.